### ISSUE TITLE
Is it possible to enhance Spatiocyte computation using a computer cluster? 

### ISSUE THREAD

**Issue by CiaranWelsh** [ecell/ecell4#253]
Hi, 

Is it possible to run a spatiocyte model on a computer cluster (for example running SGE or Slurm job scheduler) and get significant speed gains? Would you be able to just run a simulation on the cluster and automatically have computation times enhanced or would you need to make adjustments to make use of the additional computing power? 

If this is possible I would find it very useful to see an example in the docs.  

Thanks, Ciaran

**Comment by kaizu** [comment]
`ecell4` only supports parallelization of multiple tasks, but not acceleration of a single task now.

When you need to run multiple jobs, SGE, Slurm, multiprocessing, and Azure Batch are available.
For example, if you want to run a single model multiple times and take the average of their time courses, `ecell4.extra.ensemble.ensemble_simulations` might be useful.
 
```python
from ecell4 import *
from ecell4.extra import ensemble

with reaction_rules():
    A + B == C | (0.01, 0.3)

ensemble.ensemble_simulations(
    10.0, {'C': 60}, solver='gillespie', return_type='matplotlib',
    n=30, method='multiprocessing')
```

`ecell4.extra.ensemble` also provides more configurable utilities like `run_slurm`, but it's not well-documented yet. I'll prepare examples, if the utility is what you need.
https://github.com/ecell/ecell4/blob/master/python/lib/ecell4/extra/ensemble.py#L167

**Comment by CiaranWelsh** [comment]
Yes this will be useful. I can make use of both slurm and SGE - Its cool that you already support both. I'll be able to work out how to use them from the code you pointed me too. Thanks!

**Comment by kaizu** [comment]
@CiaranWelsh, I added some documents about how to use `run_slurm` and others:
https://github.com/ecell/ecell4/blob/master/python/lib/ecell4/extra/ensemble.py#L325-L390

I will shortly add practical examples running `ecell4` simulations in parallel with `run_slurm`.

This issue was closed and moved to #258.
